tests/agent_features/test_super_agent_health_check.py

```diff
@@ -0,0 +1,215 @@
+
+    # Assert on contents of health file
+    assert len(contents) == 5
+    assert contents[0] == "healthy: False\n"
+    assert contents[1] == "status: Invalid license key (HTTP status code 401)\n"
+    assert contents[2] == "start_time_unix_nano: 1234567890\n"
+    assert contents[3].startswith("status_time_unix_nano:") is True
+    assert contents[4] == "last_error: NR-APM-001\n"
+
+
+def test_no_override_on_unhealthy_shutdown(monkeypatch, tmp_path):
+    # Setup expected env vars to run super agent health check
+    monkeypatch.setenv("NEW_RELIC_SUPERAGENT_FLEET_ID", "1234")
+    file_path = tmp_path.as_uri()
+    monkeypatch.setenv("NEW_RELIC_SUPERAGENT_HEALTH_DELIVERY_LOCATION", file_path)
+
+    # Write to health YAML file
+    super_agent_instance = super_agent_health_instance()
+    super_agent_instance.start_time_unix_nano = "1234567890"
+    super_agent_instance.set_health_status("invalid_license")
+
+    # Attempt to override a previously unhealthy status
+    super_agent_instance.set_health_status("agent_shutdown")
+    super_agent_instance.write_to_health_file()
+
+    contents = get_health_file_contents(tmp_path)
+
+    # Assert on contents of health file
+    assert len(contents) == 5
+    assert contents[0] == "healthy: False\n"
+    assert contents[1] == "status: Invalid license key (HTTP status code 401)\n"
+    assert contents[4] == "last_error: NR-APM-001\n"
+
+
+def test_health_check_running_threads(monkeypatch, tmp_path):
+    running_threads = threading.enumerate()
+    # Only the main thread should be running since not super agent env vars are set
+    assert len(running_threads) == 1
+
+    monkeypatch.setenv("NEW_RELIC_SUPERAGENT_FLEET_ID", "1234")
+    file_path = tmp_path.as_uri()
+    monkeypatch.setenv("NEW_RELIC_SUPERAGENT_HEALTH_DELIVERY_LOCATION", file_path)
+
+    # Re-initialize the agent to allow the health check thread to start and assert that it did
+    _reset_configuration_done()
+    initialize()
+
+    running_threads = threading.enumerate()
+
+    assert len(running_threads) == 2
+    assert running_threads[1].name == "NR-Control-Health-Main-Thread"
+
+
+def test_proxy_error_status(monkeypatch, tmp_path):
+    # Setup expected env vars to run super agent health check
+    monkeypatch.setenv("NEW_RELIC_SUPERAGENT_FLEET_ID", "1234")
+    file_path = tmp_path.as_uri()
+    monkeypatch.setenv("NEW_RELIC_SUPERAGENT_HEALTH_DELIVERY_LOCATION", file_path)
+
+    _reset_configuration_done()
+    initialize()
+
+    # Mock a 407 error to generate a proxy error health status
+    HttpClientRecorder.STATUS_CODE = 407
+    settings = finalize_application_settings(
+        {
+            "license_key": "123LICENSEKEY",
+        }
+    )
+    protocol = AgentProtocol(settings, client_cls=HttpClientRecorder)
+
+    with pytest.raises(DiscardDataForRequest):
+        protocol.send("analytic_event_data")
+
+    # Give time for the scheduler to kick in and write to the health file
+    time.sleep(5)
```

Comment:
I don't love leaving an intentional race condition in the tests but I don't have a much better solution.